tests/supply_chain/test_action_reset.py

```diff
@@ -0,0 +1,790 @@
+                states_1_consumer[i][idx] = consumer_nodes[i:idx:consumer_features].flatten().astype(np.int)
+
+            for idx in range(len(storage_nodes)):
+                states_1_storage[i][idx] = list(storage_nodes[i:idx:storage_features].flatten().astype(np.int))
+                states_1_storage[i][idx].append(storage_nodes[i:idx:"product_id_list"].flatten().astype(np.int))
+                states_1_storage[i][idx].append(storage_nodes[i:idx:"product_quantity"].flatten().astype(np.int))
+                states_1_storage[i][idx].append(storage_nodes[i:idx:"remaining_space"].flatten().astype(np.int))
+
+            for idx in range(len(manufacture_nodes)):
+                states_1_manufacture[i][idx] = manufacture_nodes[i:idx:manufacture_features].flatten().astype(np.int)
+
+            for idx in range(len(distribution_nodes)):
+                states_1_distribution[i][idx] = distribution_nodes[i:idx:distribution_features].flatten().astype(np.int)
+
+            for idx in range(len(seller_nodes)):
+                states_1_seller[i][idx] = seller_nodes[i:idx:seller_features].flatten().astype(np.int)
+
+        # ############################### Test whether reset updates the storage unit completely ################
+        env.reset()
+        env.step(None)
+
+        # snapshot should reset after env.reset().
+        for idx in range(len(manufacture_nodes)):
+            states = manufacture_nodes[1:idx:manufacture_features].flatten().astype(np.int)
+            self.assertEqual([0, 0, 0, 0, 0, 0, 0], list(states))
+
+        for idx in range(len(storage_nodes)):
+            states = storage_nodes[1:idx:storage_features].flatten().astype(np.int)
+            self.assertEqual([0, 0], list(states))
+
+        for idx in range(len(distribution_nodes)):
+            states = distribution_nodes[1:idx:distribution_features].flatten().astype(np.int)
+            self.assertEqual([0, 0, 0, 0], list(states))
+
+        for idx in range(len(consumer_nodes)):
+            states = consumer_nodes[1:idx:consumer_features].flatten().astype(np.int)
+            self.assertEqual([0, 0, 0, 0, 0, 0, 0, 0, 0], list(states))
+
+        for idx in range(len(seller_nodes)):
+            states = seller_nodes[1:idx:seller_features].flatten().astype(np.int)
+            self.assertEqual([0, 0, 0, 0, 0, 0, 0, 0], list(states))
+
+        expect_tick = 10
+
+        # Save the env.metric of each tick into env_metric_2
+        env_metric_2: Dict[int, dict] = defaultdict(dict)
+
+        # Store the information about the snapshot storage unit of each tick in states_2
+
+        states_2_consumer: Dict[int, dict] = defaultdict(dict)
+        states_2_storage: Dict[int, dict] = defaultdict(dict)
+        states_2_seller: Dict[int, dict] = defaultdict(dict)
+        states_2_manufacture: Dict[int, dict] = defaultdict(dict)
+        states_2_distribution: Dict[int, dict] = defaultdict(dict)
+
+        for i in range(expect_tick):
```

Comment:
- you can wrap line 110-130 as a function, so as to reduce the redundant code(current line 170-190, similar parts in other test functions)

- snapshot can be accessed (all ticks) only once at the end of the simulation.